### PR TITLE
fix: blocks do no merge when moved

### LINKED ISSUE #3: Blocks merge when displaced in quick succession
image

Press 1 and 2, in quick succession, blocks will merge.

### PR DESCRIPTION
**Pull request by nicholaschuayunzhi**
fixes #3
Uses raycast to check if block can move to desired location

### CODE REVIEW

**Comment by Rinder5** [comment]
I believe this fixes #3 instead =P

File: Assets/Scripts/BlockBehaviour.cs [hunk 1]
```diff
@@ -27,21 +41,28 @@ private void Update()
...
+    private BlockFaceBehaviour GetOppositeFace(BlockFaceBehaviour face)
+    {
+        return
+            (face == Top) ? Bottom :
+            (face == Bottom) ? Top :
+            (face == Front) ? Back :
+            (face == Back) ? Front :
+            (face == Left) ? Right :
+            (face == Right) ? Left : null;
+    }
```
Comment (Rinder5) [inline Assets/Scripts/BlockBehaviour.cs:53]: I like your improv switch statement. =)

File: Assets/Scripts/BlockBehaviour.cs [hunk 2]
```diff
@@ -5,12 +5,26 @@
...
+     */
+    public BlockFaceBehaviour Top;
+    public BlockFaceBehaviour Bottom;
+    public BlockFaceBehaviour Front;
+    public BlockFaceBehaviour Back;
+    public BlockFaceBehaviour Left;
+    public BlockFaceBehaviour Right;
+
+    [Range(0f, 1f)]
+    public float SkinToLengthRatio = 0.1f;
```
Comment (Rinder5) [inline Assets/Scripts/BlockBehaviour.cs:19]: Why is it called skin? Is there more descriptive name?

**Review by Rinder5 — COMMENTED** [review]
Note: does not actually fix bug, but minimizes it to an acceptable level. Good job!

File: Assets/Scripts/BlockBehaviour.cs (hunk 2 above)
Comment (nicholaschuayunzhi) [reply Assets/Scripts/BlockBehaviour.cs:19]: Raycast are fired from within the blocks. Thus we are firing from under the block's "skin".

**Pull request merged**